libs/designsystem/src/lib/components/popover/popover.component.ts

```diff
@@ -0,0 +1,210 @@
+    this.hide();
+  }
+
+  // document.removeEventListener needs the exact same event handler & options reference:
+  private readonly preventScrollEventListenerOptions = {
+    passive: false,
+  } as EventListenerOptions;
+
+  // document.removeEventListener needs the exact same event handler & options reference:
+  private preventEvent(event: TouchEvent) {
+    event.preventDefault();
+  }
+
+  private preventScroll() {
+    this.renderer.addClass(this.document.body, 'backdrop-no-scroll'); // TODO use our own class instead of relying on Ionic?
+    this.document.addEventListener(
+      'touchmove',
+      this.preventEvent,
+      this.preventScrollEventListenerOptions
+    );
+  }
+
+  private releaseScroll() {
+    this.renderer.removeClass(this.document.body, 'backdrop-no-scroll'); // TODO use our own class instead of relying on Ionic?
+    this.document.removeEventListener(
+      'touchmove',
+      this.preventEvent,
+      this.preventScrollEventListenerOptions
+    );
+  }
+
+  show() {
+    this.renderer.addClass(this.elementRef.nativeElement, 'is-opening');
+    this.renderer.appendChild(this.document.body, this.elementRef.nativeElement);
+
+    this.positionWrapper();
+    this.preventScroll();
+
+    this.renderer.addClass(this.elementRef.nativeElement, 'is-open');
+    this.renderer.removeClass(this.elementRef.nativeElement, 'is-opening');
+
+    this.isShowing = true;
+  }
+
+  hide() {
+    this.renderer.removeChild(this.document.body, this.elementRef.nativeElement);
+    this.releaseScroll();
+
+    this.renderer.removeStyle(this.targetElement, 'z-index');
+    this.renderer.removeStyle(this.targetElement, 'pointer-events');
+    this.isShowing = false;
+  }
+
+  private readonly POPOVER_BODY_PADDING = 12;
+  private positionWrapper() {
+    const targetDimensions = this.targetElement.getBoundingClientRect();
+
+    this.renderer.addClass(this.elementRef.nativeElement, 'is-opening');
+    this.renderer.appendChild(this.document.body, this.elementRef.nativeElement);
+
+    const popoverElement = this.wrapperElement.nativeElement;
+    const popoverDimensions = popoverElement.getBoundingClientRect();
+    const viewport = popoverElement.ownerDocument.defaultView;
+    this.positionHorizontally(viewport, targetDimensions, popoverDimensions);
+    this.positionVertically(viewport, targetDimensions, popoverDimensions);
+  }
+
+  private positionHorizontally(
+    viewPort: Window,
+    targetDimensions: DOMRect,
+    popoverDimensions: DOMRect
+  ) {
+    const viewPortWidth = viewPort.innerWidth;
+    const popoverElement = this.wrapperElement.nativeElement;
+    const contentWidth = popoverDimensions.width;
+    const availableSpaceRight = viewPortWidth - targetDimensions.left;
+    const availableSpaceLeft = targetDimensions.right;
+    const contentCanRightOfTarget = availableSpaceRight >= contentWidth + this.POPOVER_BODY_PADDING;
+    const openRight = contentCanRightOfTarget || availableSpaceRight >= availableSpaceLeft;
```

Comment:
Is a bit hard to understand why the last part exists `|| availableSpaceRight >= availableSpaceLeft;` -> is there a way to make it more readable? 

Note: it exists to open right in the case where there is not enough space to open right, but there is more space to the right than the left; in that case we want to open right to make the most use of available space! huzzah!